memecity.game/DrawableComponent.h

```diff
@@ -0,0 +1,18 @@
+﻿#ifndef _DRAWABLECOMPONENT_H
+#define _DRAWABLECOMPONENT_H
+
+#include "Texture.h"
+#include "../memecity.engine.ecs/Component.h"
+
+struct DrawableComponent : public Component
+{
+	static std::string COMPONENT_TYPE;
+
+public:
+	std::string get_type() override;
+	std::shared_ptr<Texture>  texture;
```

Comment:
Why is this a shared_ptr?

Reply:
Because the multimedia manager returns a shared_ptr, and a weak_ptr is not applicable as DrawableComponent becomes the owner of the texture.